src/http/common.cr

```diff
@@ -53,10 +53,14 @@ module HTTP
         end
 
         if decompress && body
+          encoding = headers["Content-Encoding"]?
           {% if flag?(:without_zlib) %}
-            raise "Can't decompress because `-D without_zlib` was passed at compile time"
+            case encoding
+            when "gzip"
```

Comment:
What's this for?

Reply:
https://github.com/crystal-lang/crystal/pull/8683/files#r365761503